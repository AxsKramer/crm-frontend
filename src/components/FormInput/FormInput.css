.field {
  padding: .5rem 0;
  margin: 2rem 0;
  width: 100%;
}
.field label {
  font-weight: bold;
  font-family: var(--fontBoton);
  font-size: 1.4rem;
  display: none;
}
.field input:not([type="submit"]){
  padding: 1.5rem;
  border: 1px solid #e1e1e1;
  outline: none;
  font-size: 1.2rem;
  width: 100%;
}

@media only screen and (min-width: 766px){
  .field{
    display: grid;
    grid-template-columns: 30% 70%;
    align-items: center;
    justify-items: center;
  }
  .field label {
    display: block;
  }
  .field input:not([type="submit"]){
  }

}