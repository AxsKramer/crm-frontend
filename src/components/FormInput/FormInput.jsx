import React from "react";
import './FormInput.css';

const FormInput = ({label ='', type, name, handleChange, value, placeholder,id, ...otherprops }) => {
  return (
    <div className="field">
      <label htmlFor={id}>{label}</label>
      <input
        type={type}
        name={name}
        placeholder={placeholder}
        onChange={handleChange}
        id={id}
        value={value}
        {...otherprops}
      />
    </div>
  );
};

export default FormInput;
