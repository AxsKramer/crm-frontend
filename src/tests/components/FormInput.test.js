import React from 'react';
import {shallow} from 'enzyme';
import FormInput from '../../components/FormInput/FormInput';

// import '@testing-library/jest-dom';

describe("Testing of <FormInput />", () => {

  const label = 'Project';
  const type = 'text';
  const name = 'project';
  const handleChange = () => {};
  const value = 'MERN';
  const placeholder = 'Project Name';
  const id = 1;
  
  const wrapper = shallow(<FormInput label={label} type={type} name={name} handleChange={handleChange} value={value} placeholder={placeholder} id={id} />);

  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should be only one component to render', () => {
    expect(wrapper.length).toEqual(1);
  })

  test('should have a label', () => {
    const label = wrapper.find('label').length;
    expect(label).toEqual(1);
  })
  
  test('text label should be equal to "Project" ', () => {
    const labelText = wrapper.find('label').text();
    expect(labelText).toEqual("Project")
  })

  test('should have MERN as value', () => {
    const inputValue = wrapper.find('input');
    expect(inputValue.prop('value')).toBe('MERN');
  })

  test('input type can not be submit ', () => {
    const inputType = wrapper.find('input').prop('type');
    expect(inputType).not.toBe('submit');
  })

  test('div should have a class name "field"', () => {
    const className = wrapper.find('div').prop('className');
    expect(className.includes('field')).toBe(true);
  })
  
})
