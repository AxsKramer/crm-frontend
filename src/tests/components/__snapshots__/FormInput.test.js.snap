// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of <FormInput /> should be displayed correctly 1`] = `
<div
  className="field"
>
  <label
    htmlFor={1}
  >
    Project
  </label>
  <input
    id={1}
    name="project"
    onChange={[Function]}
    placeholder="Project Name"
    type="text"
    value="MERN"
  />
</div>
`;
